program-runtime/src/invoke_context.rs

```diff
@@ -523,6 +525,19 @@ impl<'a> InvokeContext<'a> {
             let owner_id = borrowed_root_account.get_owner();
             if native_loader::check_id(owner_id) {
                 *borrowed_root_account.get_key()
+            } else if self
+                .get_feature_set()
+                .is_active(&remove_accounts_executable_flag_checks::id())
+            {
+                if bpf_loader_deprecated::check_id(owner_id)
+                    || bpf_loader::check_id(owner_id)
+                    || bpf_loader_upgradeable::check_id(owner_id)
+                    || loader_v4::check_id(owner_id)
```

Comment:
Shouldn't we apply the same checks to account loading? We could get rid of all of the `validated_loaders` logic there.

Reply:
Haha, great, follow up PR then? And these changes should be amended to the SIMD: https://github.com/solana-foundation/solana-improvement-documents/pull/162